src/components/Login/accountConnection/index.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  Button,
+  Flex,
+  Icon,
+  Link,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Text,
+} from "@chakra-ui/react";
+import React, { Dispatch, SetStateAction } from "react";
+import MainLogoSvg from "@/assets/svg/CourtCanva-main-LOGO.svg";
+import { FcGoogle } from "react-icons/fc";
+import ModalOperator from "@/components/Login/ModalOperater";
+import { api } from "@/utils/axios";
+import { GoogleLoginRes } from "@/components/Login/SelectLogin";
+import { updateUserInfo } from "@/store/reducer/userSlice";
+import { useStoreDispatch } from "@/store/hooks";
+
+interface Props {
+  existedUserInfo: GoogleLoginRes | null;
+  setStep: Dispatch<SetStateAction<number>>;
+  updateLoginData: (data: GoogleLoginRes) => void;
+  onClose: () => void;
+}
+
+const AccountConnection: React.FC<Props> = ({
+  existedUserInfo,
+  setStep,
+  updateLoginData,
+  onClose,
+}) => {
+  const dispatch = useStoreDispatch();
+  const onConnect = async () => {
+    const { data } = await api("/user/connect", {
+      method: "put",
+      requestData: {
+        googleId: existedUserInfo?.googleId,
+        email: existedUserInfo?.email,
+      },
+    });
+    try {
+      if (data) {
+        localStorage.setItem("UserInfo", JSON.stringify(data));
+        dispatch(updateUserInfo(data));
+        updateLoginData(data);
+        setStep(1);
+        onClose();
+      }
+    } catch (err) {
+      console.warn(err);
+    }
+  };
```

Comment:
Are we going to have this connection API called only here or maybe somewhere else in the future? If only here then all good, otherwise placing it into redux might be a better way. Just asking.

Reply:
Thank you 桃总, for now I haven't come up with a situation where this call is also needed, I'll move it to Redux if there's a need for it in the future :).

Reply:
> Are we going to have this connection API called only here or maybe somewhere else in the future? If only here then all good, otherwise placing it into redux might be a better way. Just asking.

Thank you 桃总, for now I haven't come up with a situation where this call is also needed, I'll move it to Redux if there's a need for it in the future :).

Reply:
You can leave a comment to describe it just in case.